src/mutate.rs

```diff
@@ -102,8 +118,92 @@ impl Client {
             .await?
             .into_inner();
 
-        self.collection_cache.update_timestamp(&collection_name, result.timestamp);
+        self.collection_cache
+            .update_timestamp(&collection_name, result.timestamp);
+
+        Ok(result)
+    }
+
+    pub async fn delete(
+        &self,
+        collection_name: impl Into<String>,
+        options: &DeleteOptions,
+    ) -> Result<crate::proto::milvus::MutationResult> {
+        if options.ids.len() > 0 && options.filter.len() > 0 {
```

Comment:
maybe we can do something when constrcuting DeleteOptions to avoid set both of `ids` and `filter`? like only offer `with_ids` and `with_filter` to set these fields.